vital/iterator.h

```diff
@@ -477,6 +478,102 @@ class const_iterator
   }
 };
 
+/**
+ * Pure-virtual mixin class to add iteration support to a class.
+ *
+ * \tparam Type to iterate over.
+ */
+template< typename T >
+class iterable
+{
+public:
+  using iterator       = vital::iterator< T >;
+  using const_iterator = vital::const_iterator< T >;
+
+  /// Constructor
+  iterable() = default;
+  /// Destructor
+  virtual ~iterable() = default;
+
+  /** @name Iterator Accessors
+   * Accessors for const and non-const iterators
+   */
+  ///@{
+  /**
+   * Get the non-const iterator to the beginning of the collection.
+   * @return An iterator over the objects in this collection.
+   */
+  virtual iterator begin()
+  {
+    return vital::iterator< T >( get_iter_next_func() );
+  }
+
+  /**
+   * Get the non-const iterator past the end of the collection
+   * @return An iterator base the end of this collection.
+   */
+  virtual iterator end()
+  {
+    return vital::iterator< T >();
+  }
+
+  /**
+   * Get the const iterator to the beginning of the collection.
+   * @return An iterator over the objects in this collection.
+   */
+  virtual const_iterator begin() const
+  {
+    return vital::const_iterator< T >( get_const_iter_next_func() );
+  }
+
+  /**
```

Comment:
Probably just tired or blind, but I don't see the difference between the second version of begin (the const one) and the cbegin. Same goes for the second version of end and cend.

Reply:
@linus-sherrill ?